### ISSUE TITLE
RegistrationsController#edit_registrations is very slow

### ISSUE THREAD

**Issue by timhabermaas** [thewca/worldcubeassociation.org#899]
Here are screenshots from NewRelic. The data is from the last 3h and there were at most 70rpm (avg 30rpm), so it's not like our server was being hammered at any time.

<img width="753" alt="screen shot 2016-09-16 at 11 34 46 am" src="https://cloud.githubusercontent.com/assets/96054/18581644/cee4d4d0-7c01-11e6-8d43-a12ae990b681.png">

<img width="745" alt="screen shot 2016-09-16 at 11 35 10 am" src="https://cloud.githubusercontent.com/assets/96054/18581683/f382db0c-7c01-11e6-8622-c1ead12ca966.png">

`RegistrationsController#do_actions_for_selected` isn't much faster btw (average response times for the last 3h):
<img width="417" alt="screen shot 2016-09-16 at 11 38 08 am" src="https://cloud.githubusercontent.com/assets/96054/18581777/5bb4d8b0-7c02-11e6-8dce-1d49a36fcca7.png">

**Comment by viroulep** [comment]
noobie question here: do we have any tool to do profiling locally?

**Comment by timhabermaas** [comment]
AFAIK you can use NewRelic's [developer mode](https://docs.newrelic.com/docs/agents/ruby-agent/developer-mode/developer-mode) by visiting a page and afterwards going to `http://localhost:3000/newrelic`. It's usually enough to find the very slow bits. If it's not, there's always [`ruby-prof`](https://github.com/ruby-prof/ruby-prof#profiling-rails).

**Comment by viroulep** [comment]
Woops, re-opening this because there may be room for improvements with Tim's solution :)

At least the performance are decent again now.

**Comment by viroulep** [comment]
Working on it right now (the general issue with the new models' `find` calls).

**Comment by ronaldmansveld** [comment]
@jfly @viroulep Is this issue still relevant?

**Comment by viroulep** [comment]
I don't see any PR referencing it so I'd guess yes, but I'd need to look at it again.

**Comment by viroulep** [comment]
I just looked at it again and it seems to be much better than what Tim originally reported.
I had to force a version of newrelic_rpm <= 4.0 locally because the developer mode got removed starting from 4.1.

The breakdown looks much better (for a ~100 registrations competitions):
![new_relic](https://user-images.githubusercontent.com/1007485/46867531-985acf00-ce25-11e8-9199-9e67056a952b.png)

There are 40ish sql queries for the page, but all are 1 or 2 ms, so it seems fine.
